snowflake.go

```diff
@@ -135,15 +134,17 @@ func NewNode(node int64) (*Node, error) {
 func (n *Node) Generate() ID {
 
 	n.mu.Lock()
+	defer n.mu.Unlock()
 
-	now := time.Since(n.epoch).Nanoseconds() / 1000000
+	now := time.Since(n.epoch).Milliseconds()
 
 	if now == n.time {
 		n.step = (n.step + 1) & n.stepMask
 
 		if n.step == 0 {
-			for now <= n.time {
-				now = time.Since(n.epoch).Nanoseconds() / 1000000
+			now += 1 // set it to the next Millisecond
+			if time.Since(n.epoch.Add(time.Duration(now)*time.Millisecond)) < 0 {
```

Comment:
I think I understand what you're going for with this change but I'm unsure if this works out as better or not.  time.Sleep() may not be ms accurate on many systems so it's likely this will often take more time though maybe a bit less CPU than the current method.

Reply:
Thank you for sharing the story of `time.Sleep()`